DSpace/dspace-angular
# Unclickable title in search list fix

## References
* Fixes https://github.com/DSpace/DSpace/issues/3104

## Description
This PR ensures that Items in the search lists will always have a clickable title.
This is done by altering the retrieval of the first occurrence of dc.title to the call to `dsoNameService.getName(this.dso);` This method will then provide us with the name of the DSO by calling `dso.firstMetadataValue('dc.title') || dso.name || this.translateService.instant('dso.name.untitled');` and therefore, if a dc.title and a dso name don't exist, it'll return an "Untitled" String which is translatable

## Instructions for Reviewers

List of changes in this PR:
* Changed the retrieval of a dso's name in the search list from a dc.title metadata call to a more broad dsoNameService.getName call
* Changed the dsoNameService.getName call to return an "Untitled" translatable String if a DSO doesn't have a title/name
* Added tests for this behaviour

How to test this PR:
* Setup this Angular branch with the api7.dspace.org as backend
* Search for `-dc.title:*` in the search bar
* Verify that Items are returned with an "Untitled" name instead of them being unclickable

## Checklist
